xarray/core/computation.py

```diff
@@ -1933,14 +1933,18 @@ def _ensure_numeric(data: T_Xarray) -> T_Xarray:
     from .dataset import Dataset
 
     def to_floatable(x: DataArray) -> DataArray:
-        if x.dtype.kind in "mM":
+        if x.dtype.kind == "M":
+            # datetimes
             return x.copy(
                 data=datetime_to_numeric(
                     x.data,
                     offset=np.datetime64("1970-01-01"),
                     datetime_unit="ns",
                 ),
             )
+        elif x.dtype.kind == "m":
```

Comment:
LGTM.

We also have another `_floatize` 
https://github.com/pydata/xarray/blob/6bb2b855498b5c68d7cca8cceb710365d58e6048/xarray/core/missing.py#L572-L589

so it might be nice to harmonize in a future PR. For example, it seems like interp doesn't work with timedelta right now?